pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotBrokerHealthCheck.java

```diff
@@ -61,7 +61,9 @@ public String getBrokerHealth() {
       return "OK";
     }
     _brokerMetrics.addMeteredGlobalValue(BrokerMeter.HEALTHCHECK_BAD_CALLS, 1);
-    throw new WebApplicationException(String.format("Pinot broker status is %s", status),
-        Response.Status.SERVICE_UNAVAILABLE);
+    Response response =
```

Comment:
Any specific reason for this change? I think we are using the constructor from the original code in a lot of places

Reply:
I did a bit more digging cause this was bothering me. `ControllerApplicationException` uses the same constructor, but it seems it's intercepted by Jersey which adds the message to the response. Whereas the healthchecks go through jax-rs directly which doesn't build the response with the message in it. I can't quite trace how these healthcheck endpoints are created and how they're different. If you have a pointer here, that would be great.

Reply:
I tried both constructors, and seems the new one will change the message to `"HTTP 503 Service Unavailable"` and encode the actual error message as the entity of the response. Changing the response format can potentially cause backward incompatibility.
@xiangfu0 Do you have some idea on this issue?

Reply:
I'm not sure we'd actually see an incompatibility. Before you were getting no message, so I don't see how anyone could have been depending on the message. Going from empty -> non-empty should be safe.

Reply:
Here is the properties of the exception built from the 2 constructors. `exception1` is from the existing code, `exception2` is from the new code. Note that the `detailedMessages` is changed, and the new one has entity embedded.

![Screen Shot 2022-01-13 at 4 23 55 PM](https://user-images.githubusercontent.com/17555551/149430027-98b0544f-0a31-4831-b988-c7a45318e2ce.png)

I'd suggest keeping the existing behavior for consistency. If we do need to change the behavior, we should change it for all usages instead of just this one (probably in a separate PR).

Reply:
ok, i gave it one more go. if you think it should still go in a separate PR, happy to go back to the original code.

thank you for pointing me to the debugger. this new approach keeps the same detailMessage but updates the response to set the entity. This way you get the actual string message when using `curl`.